rmw_opendds_shared_cpp/include/rmw_opendds_shared_cpp/OpenDDSNode.hpp

```diff
@@ -0,0 +1,57 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
```

Comment:
The class OpenDDSNode, defined in OpenDDSNode.hpp and implemented in OpenDDSNode.cpp, was originally defined as a c-struct: OpenDDSNodeInfo in a c-style header file types.hpp (which contains several classes and structs). The functions related to OpenDDSNodeInfo were defined in multip[le c function files.

My first implementation of OpenDDSNode was made in the existing files. As more funcionalities were developed, the code became messy, and the class was moved into its own .hpp/.cpp files. At first I struggled about the file naming, and then decided to follow the OpenDDS (also generally accepted) c++ file naming convention, for these reasons:

1) To name them as opendds_node.hpp/.cpp is following the c convention, and I would have to name the files of the class RmwAllocateFree as rmw_allocate_free.hpp/.cpp, that would easily be confused with rmw function files instead of the c++ class files.

2) These classes are not exposed to applications using rmw, i.e., not RMW_OPENDDS_SHARED_CPP_PUBLIC prefixed. They are just OpenDDS implementations of some rmw function interfaces. The only difference is now we choose to use c++ class to encapsulate these implementations, instead of using the c-style types.hpp and several c function files. 

When we complete the changes for all the objects using OpenDDSNode, the c-struct OpenDDSNodeInfo and some c functions files will be removed.

Reply:
> To name them as opendds_node.hpp/.cpp is following the c convention, and I would have to name the files of the class RmwAllocateFree as rmw_allocate_free.hpp/.cpp, that would easily be confused with rmw function files instead of the c++ class files.

I don't think the lowercase_with_underscores naming is C convention. I think it's just a convention that each repo enforces. I've seen C++ repos with different file naming conventions, Google test for example uses lowercase-dash naming (https://github.com/google/googletest/tree/master/googletest/src).

Reply:
The main point is not that the lowercase is c convention. The standard c++ library class files are lowercased to match their classnames, and most custom-defined c++ classes are Camel-cased and their files also match their classnames to minimize potential conflicts. That's the simplest and straight forward way, and is adopted by most c++ practices including our OpenDDS standard. In this particular case, we are defining non-exposed OpenDDS-implementation specific classes. To name them like rmw_allocate_free.hpp/.cpp has two problems: 1) the filenames do not match the classnames and they can easily be taken as the free function files; 2) the rmw interface functions usually prefix rmw_ to the function names, and their files match their function names. The include guard usually follow the filenames in some fashion. If our implementation class RmwAllocateFree is defined in rmw_allocate_free.hpp/.cpp, it just adds more confusion with rmw files. In other words, we would be trying to conform to rmw function files for no benefits (these classes are not exposed to rmw applications), but only to confuse ourselves. In this case, we will be better off just conforming to our OpenDDS standard to match the classnames, since rmw does not care whether you implement all the rmw interface functions with your own free function files, or some c++ class files. All rmw expects is that when an rmw_ function is called, there is your specific implementation to support the functionality.

Reply:
As in my previous comment, I don't think there is a widely accepted guideline on whether one should choose CamelCase, snake_case, or other types for their C++ project. It's more of a choice of the owner of each specific project. Some projects like OpenDDS or some other DDS implementations choose CamelCase, some other projects choose snake_case, e.g., Google's C++ projects seem to choose this way (https://google.github.io/styleguide/cppguide.html#File_Names). 

Also, the file name doesn't need to match the class name it contains, for example, OpenDDS's DiscoveryBase.h doesn't have a class named DiscoveryBase. The file rmw_allocate_free.hpp isn't really a problem -- we can change it so that it doesn't have the rmw prefix if we really want to. In general, I think a single repo should name its files in a consistent way.

This isn't the main point of this PR. If you want to keep the file names, I'm ok with that.

Reply:
For reference, rmw_connext names their DDS-specific classes using CamelCase while naming the corresponding files using snake_case (for example, https://github.com/ros2/rmw_connext/blob/master/rmw_connext_cpp/src/connext_static_publisher_info.hpp).

Reply:
The style has not been clearly defined for our RMW. Let's table(create a Jira issue) these issues for now that relate to style. The only thing I have said in the past is to be consistent but at this point I don't think we're going to fix all these style issues in this PR. 

Thanks.